overlord/snapstate/snapmgr.go

```diff
@@ -157,7 +157,9 @@ func (m *SnapManager) undoPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 	snapst.Candidate = nil
+	st.Lock()
```

Comment:
Yes, that's a better approach. We're only fiddling with state here really. Let's just defer st.Unlock.